nano/nano_node/entry.cpp

```diff
@@ -1735,8 +1735,9 @@ int main (int argc, char * const * argv)
 
 			start_threads (check_pending, pending);
 
+			auto & ledger = node->ledger;
 			size_t const pending_deque_overflow (64 * 1024);
-			for (auto i (node->store.pending.begin (transaction)), n (node->store.pending.end ()); i != n; ++i)
+			for (auto i = ledger.receivable_upper_bound (transaction, 0, 0); i; i = ledger.receivable_upper_bound (transaction, i.value ().first.account, i.value ().first.hash))
```

Comment:
I see `i.value ().first.account, i.value ().first.hash)` pattern is used in quite a few places, it would be more readable and ergonomic to provide an override for upper_bound (<account, hash>) pair

Reply:
Agreed. With the rewritten iterator classes there are now done with a normal ++i.